### PR TITLE
Update clang and flags

### PR DESCRIPTION
**Pull request by jeroenooms**
Update clang to 3.8 with libcxx and compiler flags from BDR. Tested locally.

### CODE REVIEW

**Comment by eddelbuettel** [comment]
Definitely thumbs-up on the move to 3.8; the flags are probably fine too and can get adjusted if need be.

**Comment by jeroenooms** [comment]
Yay the build has [succeeded](https://hub.docker.com/r/rocker/r-devel-ubsan-clang/builds/). BTW is there any reason why the dockerhub image is called `r-devel-ubsan-clang` instead of `r-devel-san-clang`? I had difficulty finding it.

**Comment by eddelbuettel** [comment]
>  is there any reason why the dockerhub image is called r-devel-ubsan-clang instead of r-devel-san-clang?

I had several going at once with san vs ubsan, gcc vs clang, ... and tried map docker hub names to the github repo names.

**Comment by jeroenooms** [comment]
Can we rename one or the other so that they match?

**Comment by eddelbuettel** [comment]
Let me try again:

1)  I have a large set of repos I started, making the naming tough
2) The names generally match between gh and gh and when they don't I tried to improve at dh

Renaming now breaks links etc pp so unless there is a good reason (and: _I don't like the current scheme_ isn't one) I won't change it.

It is what it is.  Memory and consistency are overrated in the age of good search engines.
